### PR TITLE
Setup leap s04

### PR DESCRIPTION
**Pull request by IsaackMwenda**

### CODE REVIEW

File: code_schemes/demographics/preffered_language.json [hunk 1]
```diff
@@ -0,0 +1,207 @@
+{
+  "SchemeID": "Scheme-95fb4413",
+  "Name": "Kakuma preffered language",
```
Comment (as2388) [inline code_schemes/demographics/preffered_language.json:3]:
```suggestion
  "Name": "Kakuma Preferred Language",
```

File: code_schemes/ws_correct_dataset.json [hunk 2]
```diff
@@ -14,6 +14,17 @@
         "kakuma household language"
       ]
     },
+    {
+      "CodeID": "code-e174c918",
+      "CodeType": "Normal",
+      "DisplayText": "kakuma preffered language",
```
Comment (as2388) [inline code_schemes/ws_correct_dataset.json:20]:
```suggestion
      "DisplayText": "kakuma preferred language",
```

File: configurations/docker_image_name.txt [hunk 3]
```diff
@@ -1 +1,2 @@
-ghcr.io/africasvoices/engagement-data-pipeline:fix_telegram_uuid_formart
+ghcr.io/africasvoices/engagement-data-pipeline:v3.0.0
```
Comment (as2388) [inline configurations/docker_image_name.txt:1]: Is this project impacted by any breaking changes?

File: configurations/leap_s04_pipeline.py [hunk 4]
```diff
@@ -0,0 +1,372 @@
+from core_data_modules.cleaners import swahili
+
+from src.pipeline_configuration_spec import *
+
+
+PIPELINE_CONFIGURATION = PipelineConfiguration(
+    pipeline_name="leap_s04",
```
Comment (as2388) [inline configurations/leap_s04_pipeline.py:7]: It looks like this "wusc_leap_s04" in some other places, does it make sense to use that here too or stick with "leap_s04"? I have no strong opinions.

File: configurations/leap_s04_pipeline.py [hunk 5]
```diff
@@ -0,0 +1,372 @@
...
+    rapid_pro_sources=[
+        RapidProSource(
+            rapid_pro=RapidProClientConfiguration(
+                domain="textit.com",
+                token_file_url="gs://avf-credentials/wusc-leap-kalobeyei-textit-token.txt"
+            ),
+            sync_config=RapidProToEngagementDBConfiguration(
+                flow_result_configurations=[
+                    FlowResultConfiguration("wusc_leap_s04_kalobeyei_demogs", "preffered_language",
+                                            "kakuma_preffered_language"),
```
Comment (as2388) [inline configurations/leap_s04_pipeline.py:29]:
```suggestion
                    FlowResultConfiguration("wusc_leap_s04_kalobeyei_demogs", "preferred_language",
                                            "kakuma_preferred_language"),
```

File: configurations/leap_s04_pipeline.py [hunk 6]
```diff
@@ -0,0 +1,372 @@
...
+                    FlowResultConfiguration("wusc_leap_s04_kalobeyei_demogs", "age",
+                                            "kakuma_age"),
+                    FlowResultConfiguration("wusc_leap_s04_kalobeyei_demogs", "location",
+                                            "kakuma_location"),
+                    FlowResultConfiguration("wusc_leap_s04_kalobeyei_demogs", "nationality",
+                                            "kakuma_nationality"),
+                    FlowResultConfiguration("wusc_leap_s04_kalobeyei_demogs", "gender",
+                                            "kakuma_gender"),
+                    FlowResultConfiguration("wusc_leap_s04_kalobeyei_demogs", "disability",
+                                            "kakuma_disabled"),
```
Comment (as2388) [inline configurations/leap_s04_pipeline.py:39]: Is the kakuma_disabled field new or used before?

(no need to say in the code, just curious here)

File: configurations/leap_s04_pipeline.py [hunk 7]
```diff
@@ -0,0 +1,372 @@
...
+    rapid_pro_target=RapidProTarget(
+        rapid_pro=RapidProClientConfiguration(
+            domain="textit.com",
+            token_file_url="gs://avf-credentials/wusc-leap-kalobeyei-textit-token.txt"
+        ),
+        sync_config=EngagementDBToRapidProConfiguration(
+            normal_datasets=[
+                DatasetConfiguration(
+                    engagement_db_datasets=["kakuma_preffered_language"],
+                    rapid_pro_contact_field=ContactField(key="pool_wusc_age", label="pool wusc age")
```
Comment (as2388) [inline configurations/leap_s04_pipeline.py:330]: language or age?

File: configurations/leap_s04_pipeline.py [hunk 8]
```diff
@@ -0,0 +1,372 @@
...
+                    engagement_db_datasets=["kakuma_gender"],
+                    rapid_pro_contact_field=ContactField(key="pool_wusc_gender", label="pool wusc gender")
+                ),
+                DatasetConfiguration(
+                    engagement_db_datasets=["kakuma_location"],
+                    rapid_pro_contact_field=ContactField(key="pool_wusc_location", label="pool wusc location")
+                ),
+                DatasetConfiguration(
+                    engagement_db_datasets=["kakuma_nationality"],
+                    rapid_pro_contact_field=ContactField(key="pool_wusc_disabled", label="pool wusc nationality")
```
Comment (as2388) [inline configurations/leap_s04_pipeline.py:342]: disabled or nationality?

File: configurations/leap_s04_pipeline.py [hunk 9]
```diff
@@ -0,0 +1,372 @@
...
+            ),
+        ],
+        ws_correct_dataset_code_scheme=load_code_scheme("ws_correct_dataset"),
+    ),
+    rapid_pro_target=RapidProTarget(
+        rapid_pro=RapidProClientConfiguration(
+            domain="textit.com",
+            token_file_url="gs://avf-credentials/wusc-leap-kalobeyei-textit-token.txt"
+        ),
+        sync_config=EngagementDBToRapidProConfiguration(
```
Comment (as2388) [inline configurations/leap_s04_pipeline.py:326]: Is the plan to run this sync continuously?

**Review by as2388 — APPROVED** [review]
Approved but with a q on continuous sync back to textit

**Review by DanielMwendwa — APPROVED** [review]

**Pull request merged**